lib/iruby/command.rb

```diff
@@ -46,69 +80,27 @@ def check_version
     def run_ipython
       check_version
 
-      dir = @args.grep(/\A--iruby-dir=.*\Z/)
-      @args -= dir
-      dir = dir.last.to_s.sub(/\A--profile=/, '')
-      dir = ENV['IRUBYDIR'] || IRUBYDIR if dir.empty?
-      dir = File.expand_path(dir)
-      ENV['IPYTHONDIR'] = dir
-
-      if @args.size == 3 && @args[0] == 'profile' && @args[1] == 'create'
-        profile = @args[2]
-      else
-        profile = @args.grep(/\A--profile=.*\Z/).last.to_s.sub(/\A--profile=/, '')
-        profile = 'default' if profile.empty?
-      end
-
-      create_profile(dir, profile)
-
       # We must use the console to launch the whole 0MQ-client-server stack
-      @args = %w(console --no-banner) + @args if @args.first.to_s !~ /\A\w+\Z/
+      @args = %w(console --no-banner) + @args if @args.first.to_s !~ /\A\w/
+      register_iruby_kernel if %w(console qtconsole notebook).include? @args.first
```

Comment:
Could you check here that the registration is only done if the kernel.json file doesn't exist already?